internal/pkg/provider/chain/chain_test.go

```diff
@@ -61,18 +61,39 @@ func TestGetUpgrades(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "MultiplePassed",
+			proposals: v1.Proposals{
+				newProposal(t, 1, 100, v1.StatusPassed),
+				newProposal(t, 2, 200, v1.StatusPassed),
+			},
+			expected: []*urproto.Upgrade{
+				{
+					Height: 200,
+					Type:   urproto.UpgradeType_GOVERNANCE,
+					Status: urproto.UpgradeStatus_ACTIVE,
+					Source: urproto.ProviderType_CHAIN,
+				},
+				{
+					Height: 100,
+					Type:   urproto.UpgradeType_GOVERNANCE,
+					Status: urproto.UpgradeStatus_CANCELLED,
+					Source: urproto.ProviderType_CHAIN,
+				},
+			},
+		},
 		{
 			name: "DuplicateProposalsWithPassedStatus",
 			proposals: v1.Proposals{
 				newProposal(t, 1, 100, v1.StatusPassed),
 				newProposal(t, 2, 100, v1.StatusPassed),
-				newProposal(t, 3, 200, v1.StatusPassed),
+				newProposal(t, 3, 200, v1.StatusDepositPeriod),
```

Comment:
I needed to change the status here as the newly added logic would cancel the proposal at height `100` seeing a passed upgrade at `200`, which is the correct behaviour.